src/utils.js

```diff
@@ -34,8 +34,9 @@ class HtmlElementTypeError extends Error {
 
 function checkHtmlElement(htmlElement, ...args) {
   if (
-    !(htmlElement instanceof HTMLElement) &&
-    !(htmlElement instanceof SVGElement)
+    !htmlElement.ownerDocument &&
+    !(htmlElement instanceof htmlElement.ownerDocument.HTMLElement) &&
+    !(htmlElement instanceof htmlElement.ownerDocument.SVGElement)
```

Comment:
Should this condition be this instead?

```js
  if (
    !htmlElement.ownerDocument ||
    (
      !(htmlElement instanceof htmlElement.ownerDocument.HTMLElement) &&
      !(htmlElement instanceof htmlElement.ownerDocument.SVGElement)
    )
  ) {
    throw new HtmlElementTypeError(htmlElement, matcherFn, context, validTypes)
  }
```

If there's no `htmlElement.ownerDocument`, it makes no sense to go and evaluate the rest of the condition, because the rest of the conditional depends entirely on `htmlElement.ownerDocument` not being undefined.

Please, let me know if I'm correct or not so far. If I am, I have more things to say, but I wanted to confirm up to this point at least

cc @sheerun

Reply:
Ok, here it is the additional part of what I wanted to say.

I think the above is never surfaced because `checkHtmlElement` is coincidentally always called with a valid html element that has an `ownerDocument`.

So we never really get to the point where the `instanceof` parts of the conditions are evaluated even. The thing is, when I fixed the condition locally, I get that `htmlElement.ownerDocument.HTMLElement` is undefined. The same with `SVGElement`.

This is the point where I'm no longer sure how to continue this fix. Care to chime in @sheerun?